### PR TITLE
fix(lbp): validate calculated asset_in amount

### PR DESCRIPTION
**Pull request by Roznovjak**

### CODE REVIEW

**Review by lumir-mrkva — APPROVED** [review]

File: pallets/lbp/src/tests.rs [hunk 1]
```diff
@@ -2372,6 +2372,59 @@ fn buy_should_work() {
...
+				amount: 5_560_301,
+				buy_price: 10_000_000,
+				fee_asset: KUSD,
+				fee_amount: 20_000,
+			}
+			.into(),
+		]);
+	});
+}
+
```
Comment (dmoka) [inline pallets/lbp/src/tests.rs:2427]: I think we should also add negative test cases triggering the `InsufficientAssetBalance` error, both when we buy accumulated and distributed assets.

**Review by dmoka — APPROVED** [review]
Apart from some missing test cases, it looks fine!

If you can add those, that would be great, then we can merge it back.

File: pallets/lbp/src/lib.rs [hunk 2]
```diff
@@ -1157,6 +1153,11 @@ impl<T: Config> AMM<T::AccountId, AssetId, AssetPair, BalanceOf<T>> for Pallet<T
 				Error::<T>::MaxInRatioExceeded
 			);
 
+			ensure!(
```
Comment (dmoka) [inline pallets/lbp/src/lib.rs:1156]: I think we should also add negative test cases triggering the InsufficientAssetBalance error, both when we buy accumulated and distributed assets.

**Review by dmoka — APPROVED** [review]

File: pallets/lbp/src/tests.rs (hunk 1 above)
Comment (Roznovjak) [reply pallets/lbp/src/tests.rs:2441]: I have just added some two missing cases. Please check the latest commit.

**Pull request merged**